src/__tests__/definitions/valid-math-add.json

```diff
@@ -0,0 +1,14 @@
+{
+  "Comment": "MathAdd arguments must be paths or numbers",
+  "StartAt": "validate input",
+  "States": {
+    "validate input": {
+      "Type": "Pass",
+      "Parameters": {
+        "foo.$": "States.MathAdd(States.ArrayLength($$.Execution.Input.machines),-1)"
```

Comment:
this was failing because `asl-path-validator` was not accepting negative numbers for args in the grammar. Simple oversight which I corrected.

I also modified the grammar a bit for `States.MathAdd` to ensure the args are either a path or a number and now correctly reject the invalid case added above where `'-1'` is passed